analysis/getEfficienciesLbL.cpp

```diff
@@ -60,6 +64,16 @@ int main()
     hists[name+"_den"]->Sumw2();
   }
  
+ 
+  map<string, TH1D*> hists_eta;
+  for(auto name : histParams_eta){
```

Comment:
Suggestion (feel free to ignore): there could be just one map of histograms, just one vector of histParams and therefore one for loop to initialize all of them. The only thing that's different is binning, so one could check if name contains some key, e.g. "_vs_eta" and adjust accordingly. Something like this:

for(string name : histParams){

  if(name.find("vs_eta") != string::npos){
     hists[name]               = new TH1D( name.c_str()        ,  name.c_str()        , 24, -2.4, 2.4);
     hists[name+"_num"] = new TH1D((name+"_num").c_str(), (name+"_num").c_str(), 24, -2.4, 2.4);
     hists[name+"_den"]  = new TH1D((name+"_den").c_str(), (name+"_den").c_str(), 24, -2.4, 2.4);
  }
  else{
    hists[name]                = new TH1D( name.c_str()        ,  name.c_str()        , 9, bins);
    hists[name+"_num"]  = new TH1D((name+"_num").c_str(), (name+"_num").c_str(), 9, bins);
    hists[name+"_den"]   = new TH1D((name+"_den").c_str(), (name+"_den").c_str(), 9, bins);
  }

  hists[name+"_num"]->Sumw2();
  hists[name+"_den"]->Sumw2();
}

Reply:
Used this suggestion.